front/src/components/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,319 @@
+          fetch();
+        }
+        else {
+          setError('Server Error', 'Failed to create client');
+        }
+      })
+      .catch(error =>
+        setError('Server Error', error.message)
+      );
+    setNewClient(DEFAULT_EXTERNAL_CLIENT);
+    setTaskId(-1);
+  };
+
+  const handleClientDelete = useCallback(
+    (externalClientId: number) => {
+      deleteClient(externalClientId)
+        .then(response => {
+          if (response.ok) {
+            setSuccess('Success', `Client <ID ${externalClientId}> is deleted`);
+            fetch();
+          }
+          else {
+            setError('Server Error', 'Failed to delete client');
+          }
+        })
+        .catch(error =>
+          setError('Server Error', error.message)
+        );
+    },
+    [deleteClient, setError, setSuccess, fetch]
+  );
+
+  const taskOptions = [
+    <MenuItem key={-1} value={-1}>Unset</MenuItem>,
+    ...(tasks || []).map((task) => (
+      <MenuItem key={task.id} value={task.id}>{task.name}</MenuItem>
+    ))
+  ];
+
+  const clientsListRender = useMemo(
+    () => {
+      if (externalClients === null) {
+        return (
+          <>
+            <Typography>Loading external clients...</Typography>
+            <LinearProgress variant='query' />
+          </>
+        );
+      }
+      if (externalClients.length === 0) {
+        return <Typography variant='subtitle1'>Found no active external client</Typography>;
+      }
+      return <Paper
+        style={{
+          padding: '16px',
+          overflowX: 'auto'
+        }}
+      >
+        <Typography variant='h6'>
+          {
+            externalClients.length <= 1
+              ? 'Active external client'
+              : 'Active external clients'
+          }
+        </Typography>
+        <Table style={{
+          maxWidth: '100%',
+          overflowX: 'auto'
+        }}>
+          <TableHead>
+            <TableRow>
+              <TableCell component='th'>Name</TableCell>
+              <TableCell component='th'>Token</TableCell>
+              <TableCell component='th'>Description</TableCell>
+              <TableCell component='th'>Created By</TableCell>
+              <TableCell component='th'>Linked Task</TableCell>
+              <TableCell component='th' />
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {externalClients.map((item, i) => (
+              <TableRow
+                key={i}
+              >
+                <TableCell
+                  style={{
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
+                    maxWidth: '40ch'
+                  }}
+                  title={item.name}
+                >{item.name}</TableCell>
+                <TableCell>{item.token}</TableCell>
+                <TableCell>{item.description}</TableCell>
+                <TableCell>{item.createdBy?.name}</TableCell>
+                <TableCell>{item.task?.name || ''}</TableCell>
```

Comment:
This text should be a link pointing to the task.

Reply:
Linked